### ISSUE TITLE
feat(bundler): Allow bundler to conditionally omit legacy assets.

### ISSUE THREAD

**Issue by Harrison-Lara** [americanexpress/one-app-cli#380]
# 💡 Feature request

## Is your feature request related to a problem? Please describe

Currently, there is not a way to opt out of generating the `legacy` asset when building with one app bundler.
Build time for prod, watch:build and development is slower when generating the entire build collection produced today.
Including the `legacy` asset increases our bundle size because it includes the `legacy` bundle which does not always need to be supported for every project which can cause your module to exceed the bundle size limit even when that `legacy` bundle is never used.

## Example

Stats generated from one of our modules.

**Production Build Time:** 11 seconds `modern` asset, 16-17 seconds with `legacy` asset.
**Development Build Time:** 16 seconds `modern` asset, 23-25 seconds with `legacy` asset.
**watch:build** Time: ~3-4 second reduction.

## Describe the solution you'd like

Have the ability to pass a flag to the build command which will skip the `legacy` asset generation. Output will be: `.node`, `.browser`, `.manifest`.
An invalid flag or no flag will generate the entire asset bundle with `.node`, `.browser`, `.legacyBrowser`, `.manifest`.

## Describe alternatives you've considered

N/A

## Additional context

How to test:

Navigate to `packages/one-app-bundler`.
Run `npm pack`.
Install that package in your module.
Delete all the `.webpack-*` and `.manifest` files.
Update your package.json bundler config to use: `"bundler": { "disableLegacy": true }`
Run `npm build`.
Time the build times for `production`, `development`.
Repeat the steps, but without `"bundler": { "disableLegacy": false }`
Compare build times.

**Comment by JAdshead** [comment]
thanks for raising this. we might want to have the option of disabling the legacy bundle during development and keep for production as well as fully disabling it. One App will need to be updated to handle the case of no legacy build for a module. Its possible that there will be a mix of legacy and non legacy bundles being loaded so we will need to do testing to see how that might affect things.

**Comment by Harrison-Lara** [comment]
> thanks for raising this. we might want to have the option of disabling the legacy bundle during development and keep for production as well as fully disabling it. One App will need to be updated to handle the case of no legacy build for a module. It's possible that there will be a mix of legacy and non legacy bundles being loaded so we will need to do testing to see how that might affect things.

I agree that it would definitely be a good idea to provide the flexibility to enable and disable legacy for production, development or as a whole. As you said, there is some vetting to be done with One App to see how it handles some modules not having the legacy bundles. I have drafted the PR (#381) which can be used for testing. It allows for legacy to be disabled as an entirety. I can definitely go back and add in the options you suggested once we determine how One App handles the change.

**Comment by JAdshead** [comment]
> once we determine how One App handles the change.
One App relies on the current convention and will need to be updated to gracefully fallback to modern bundle before we can take this change, I don't see that as a quick simple change.

https://github.com/americanexpress/one-app/blob/d2f9d5c3e283a72a94cde7acc609690577703b8e/src/server/middleware/sendHtml.js#L353

**Comment by JAdshead** [comment]
we could probably take a change where it skips legacy build only in development without needing to wait for updates to one app

**Comment by Harrison-Lara** [comment]
> we could probably take a change where it skips legacy build only in development without needing to wait for updates to one app

After testing out that idea and chatting with the team about it, I think we can proceed in only allowing legacy build to be skipped for _only_ development. I've updated the [associated PR ](https://github.com/americanexpress/one-app-cli/pull/381) to reflect these changes.
